cidc_api/resources/upload_jobs.py

```diff
@@ -508,10 +554,7 @@ def poll_upload_merge_status():
     if not upload_id:
         raise BadRequest("Missing expected query parameter 'id'")
 
-    user = get_current_user()
-    upload = UploadJobs.find_by_id_and_email(upload_id, user.email)
-    if not upload:
-        raise NotFound(f"Could not find assay upload job with id {upload_id}")
+    upload = UploadJobs.find_by_id(upload_id)
```

Comment:
nit: as additional sign that `UJ.find_by_id` should not be called within `requires_upload_token_auth` - it's duplicated here, which makes this controller to issue two identical db queries for no good reason.